File: packages/muvel-api-types/src/enums/share-type.ts
export enum ShareType {
  // 비공개: 자신만 볼 수 있음
  Private,
  // 일부 공개: 링크를 통해 다른 사람과 공유 가능
  Unlisted,
  // 공개: 뮤블 내 검색을 통해 노출됨
  Public,
  // (데스크톱/모바일 앱 전용) 로컬 저장소에 저장된 소설용
  Local,
}

File: packages/tauri-app/src-tauri/src/models/enums/share_type.rs
use serde_repr::{Deserialize_repr, Serialize_repr};

#[derive(Serialize_repr, Deserialize_repr, Debug, Clone, PartialEq)]
#[repr(u8)]
pub enum ShareType {
    Private = 0,
    Unlisted = 1,
    Public = 2,
    Local = 3, // 로컬 저장용 타입
}
